models/articles-models.js

```diff
@@ -1,17 +1,21 @@
 const db = require("../db/connection");
 
-exports.selectArticle = () => {
-  return db
-    .query(
-      `SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count
-FROM articles
-LEFT JOIN comments ON comments.article_id = articles.article_id
-GROUP BY articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url
-ORDER BY articles.created_at DESC;`
-    )
-    .then((result) => {
-      return result.rows;
-    });
+exports.selectArticles = (query) => {
+  const queryValues = [];
+  let queryString =
+    "SELECT articles.author, articles.title, articles.article_id, articles.topic, articles.created_at, articles.votes, articles.article_img_url, COUNT(comments.comment_id) AS comment_count FROM articles LEFT JOIN comments ON comments.article_id = articles.article_id ";
+
+  if (query.topic) {
+    queryValues.push(query.topic);
+    queryString += "WHERE topic = $1 ";
```

Comment:
Great work adding this conditional logic and WHERE clause to your SQL query to filter by topic :)